File: 3637-trionic-array-i/3637-trionic-array-i.py
class Solution:
    def isTrionic(self, nums: List[int]) -> bool:
        p = 0
        q = len(nums) - 1
        while p < q and (nums[p + 1] > nums[p] or nums[q - 1] < nums[q]):
            if nums[p + 1] > nums[p]:
                p += 1
            if nums[q - 1] < nums[q]:
                q -= 1
        if p == 0 or q == len(nums) - 1 or p >= q:
            return False
        for i in range(p + 1, q + 1):
            if nums[i] >= nums[i - 1]:
                return False
        return True

File: 3637-trionic-array-i/3637-trionic-array-i.cpp
class Solution {
public:
    bool isTrionic(vector<int>& nums) {
        int p = 0;
        int q = nums.size() - 1;
        while(p < q && (nums[p + 1] > nums[p] || nums[q - 1] < nums[q]))
        {
            if(nums[p + 1] > nums[p])
            {
                p++;
            }
            if(nums[q - 1] < nums[q])
            {
                q--;
            } 
        }
        if(p == 0 || q == nums.size() - 1 || p >= q)
        {
            return false;
        }
        for(int i = p + 1; i <= q; i++)
        {
            if(nums[i] >= nums[i - 1])
            {
                return false;
            }
        }
        return true;
    }
};